src/pathology-report/ReportComponent.resource.ts

```diff
@@ -1,28 +1,258 @@
+  const hcPersonAttribute = person.data.results.filter(
+    (attr) => attr.attributeType.uuid === HEALTHCENTERATTRTYPEUUID
+  );
+  return hcPersonAttribute[0] ? hcPersonAttribute[0].value.uuid : null;
+}
+
+export async function getConceptAnswers(conceptUuid) {
+  const concept = await openmrsFetch(
+    `/ws/rest/v1/concept/${conceptUuid}?v=full`
+  );
+  return concept.data.answers;
+}
+
+export async function getLocations() {
+  const location = await openmrsFetch(`/ws/rest/v1/location`);
+
+  return location.data.results;
+}
+
+// export async function getPatient(patientUuid): Promise<Patient> {
+//   const searchPatient = await openmrsFetch(`/ws/rest/v1/patient/c604dabc-2700-102b-80cb-0017a47871b2?v=full`);
+//   return searchPatient.data;
 // }
+
+export async function getEncounters(): Promise<Array<EncounterResult>> {
+  const userLocationUUID = await getUserLocation();
+
+  let searchEncounter = null;
+  if (userLocationUUID) {
+    searchEncounter = await openmrsFetch(
+      `/ws/rest/v1/reportingrest/reportdata/996cf192-ff54-11eb-a63a-080027ce9ca0?location=${userLocationUUID}`
+    );
+  } else {
+    searchEncounter = await openmrsFetch(
+      `/ws/rest/v1/reportingrest/reportdata/996cf192-ff54-11eb-a63a-080027ce9ca0`
+    );
+  }
+  return searchEncounter.data.dataSets[0]
+    ? searchEncounter.data.dataSets[0].rows
+    : null;
+}
+
+export async function postSampleDropoffObs(obsObject: Obs) {
+  const ObsObjectTocreate = {
+    person: obsObject.patientUuid,
+    obsDatetime: new Date().toISOString(),
+    concept: SAMPLEDROPOFFUUID,
+    location: await getUserLocation(),
+    encounter: obsObject.encounterUuid,
+    value: YESCONCEPTUUID,
+    voided: false,
+  };
+
+  const response = await openmrsFetch("/ws/rest/v1/obs", {
+    method: "POST",
+    headers: {
+      "content-type": "application/json",
+    },
+    body: ObsObjectTocreate,
+  });
+  return response;
+}
+
+export async function voidSampleDropoff(obsObject: Obs) {
+  const ObsObjectTocreate = {
+    voided: true,
+  };
+
+  const response = await openmrsFetch(
+    `/ws/rest/v1/obs/${obsObject.sampleDropoffObsUuid}`,
+    {
+      method: "POST",
+      headers: {
+        "content-type": "application/json",
+      },
+      body: ObsObjectTocreate,
+    }
+  );
+
+  return response;
+}
+
+export async function postSampleStatusChangeObs(targetValue, obsObject: Obs) {
+  const ObsObjectTocreate = {
+    person: obsObject.patientUuid,
+    obsDatetime: new Date().toISOString(),
+    concept: SAMPLESTATUSUUID,
+    location: await getUserLocation(),
+    encounter: obsObject.encounterUuid,
+    value: targetValue.uuid,
+    voided: false,
+  };
+
+  const response = await openmrsFetch("/ws/rest/v1/obs", {
+    method: "POST",
+    headers: {
+      "content-type": "application/json",
+    },
+    body: ObsObjectTocreate,
+  });
+
+  return response;
+}
+
+export async function updateSampleStatusChangeObs(targetValue, obsObject: Obs) {
```

Comment:
Would be great to type these `targetValue` parameters

Reply:
it is an object of two values that come together.

Reply:
Sure, I mean in Typescript. Start by defining a new type or interface called `TargetValue` and maybe define it explicitly or use `extends` or the `|` operator, depending on what you actually mean by "it is an object of two values that come together."